api/src/main/java/org/xnio/XnioWorker.java

```diff
@@ -874,6 +877,12 @@ protected final int getWorkerQueueSize() {
         return taskQueue.size();
     }
 
+    public abstract XnioWorkerMXBean getMetrics();
+
+    protected abstract Closeable registerServerMetrics(XnioServerMXBean metrics);
```

Comment:
I reworked how the code works, while there still is registerServerMetric other impl changed.